awswrangler/s3/_select.py

```diff
@@ -0,0 +1,216 @@
+"""Amazon S3 Select Module (PRIVATE)."""
+
+import concurrent.futures
+import itertools
+import json
+import logging
+import pprint
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+
+from awswrangler import _utils, exceptions
+from awswrangler.s3._describe import size_objects
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+_RANGE_CHUNK_SIZE: int = int(1024 * 1024)
+
+
+def _select_object_content(
+    args: Dict[str, Any],
+    client_s3: boto3.Session,
+    scan_range: Optional[Tuple[int, int]] = None,
+) -> pd.DataFrame:
+    if scan_range:
+        args.update({"ScanRange": {"Start": scan_range[0], "End": scan_range[1]}})
+    response = client_s3.select_object_content(**args)
+
+    dfs: List[pd.DataFrame] = []
+    partial_record: str = ""
+    for event in response["Payload"]:
+        if "Records" in event:
+            records = partial_record.join(event["Records"]["Payload"].decode(encoding="utf-8", errors="ignore")).split(
+                "\n"
+            )
+            # Record end can either be a partial record or a return char
+            partial_record = records[-1]
+            dfs.append(
+                pd.DataFrame(
+                    [json.loads(record) for record in records[:-1]],
+                )
+            )
+    if not dfs:
+        return pd.DataFrame()
```

Comment:
Not a fan of this. Currently required because the scan range might return no events but we still need to return something to the loop